torch_geometric/nn/aggr/basic.py

```diff
@@ -160,9 +164,14 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
                 ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
                 dim: int = -2) -> Tensor:
 
+        if self.channels > 1:
+            self.assert_two_dimensional_input(x, dim)
+
         alpha = x
         if not isinstance(self.t, (int, float)) or self.t != 1:
```

Comment:
Is this `if` bugged? It looks like `self.t` is never int or float. Maybe condition based on
```python
if not self.learn and self._init_t !=  1.:::
```

Reply:
I think it is a float if the `learn` parameter is set to `False`
```
self.t = Parameter(torch.Tensor(self.channels)) if learn else t
```
but I think it could be simplified for clarity to 
```
t = self.t
if is tensor:
   manipulate shape
alpha = x * t
```
Not sure why we  want to avoid the multiplication by 1